wled00/data/index.js

```diff
@@ -579,10 +579,11 @@ function populateSegments(s)
 				<span class="checkmark schk"></span>
 			</label>
 			<div class="segname" onclick="selSegEx(${i})">
-				Segment ${i}
+				${inst.n ? inst.n : "Segment "+i}
 			</div>
 			<i class="icons e-icon flr ${expanded[i] ? "exp":""}" id="sege${i}" onclick="expand(${i})">&#xe395;</i>
 			<div class="segin ${expanded[i] ? "expanded":""}" id="seg${i}">
+				<input type="text" class="ptxt noslide" id="seg${i}t" autocomplete="off" maxlength=32 value="${inst.n?inst.n:""}" placeholder="Enter name..."/>
```

Comment:
This is missing in the "Add segment" div built by `makeSeg()`. This causes a JS exception in the first line of `setSeg()` upon trying to save the new segment.

Reply:
Uh! I tried to port my rewrite of segment handling (which is a bit different). Perhaps I missed something.

Reply:
My functions for segments look like this:
```javascript
function makeSeg()
{
	var ns = 0;
	if (lowestUnused > 0) {
		var pend = parseInt(gId(`seg${lowestUnused -1}e`).value,10) + (cfg.comp.seglen?parseInt(gId(`seg${lowestUnused -1}s`).value,10):0);
		if (pend < ledCount) ns = pend;
	}
	gId('segutil').scrollIntoView({
		behavior: 'smooth',
		block: 'start',
	});
	var cn = `<div class="seg">
	<div class="segin expanded">
		<input type="text" class="ptxt noslide" id="seg${lowestUnused}t" autocomplete="off" maxlength=32 value="" placeholder="New segment ${lowestUnused}"/>
		<table class="segt">
			<tr>
				<td width="38%">Start LED</td>
				<td width="38%">${cfg.comp.seglen?"LED count":"Stop LED"}</td>
			</tr>
			<tr>
				<td><input class="noslide segn" id="seg${lowestUnused}s" type="number" min="0" max="${ledCount-1}" value="${ns}" oninput="updateLen(${lowestUnused})"></td>
				<td><input class="noslide segn" id="seg${lowestUnused}e" type="number" min="0" max="${ledCount-(cfg.comp.seglen?ns:0)}" value="${ledCount-(cfg.comp.seglen?ns:0)}" oninput="updateLen(${lowestUnused})"></td>
				<td><button class="btn btn-xs" onclick="setSeg(${lowestUnused});resetUtil();"><i class="icons bth-icon" id="segc${lowestUnused}">&#xe390;</i></button></td>
			</tr>
		</table>
		<div class="h" id="seg${lowestUnused}len">${ledCount - ns} LEDs</div>
		<div class="c"><button class="btn btn-p" onclick="resetUtil()">Cancel</button></div>
	</div>
</div>`;
	gId('segutil').innerHTML = cn;
}

function resetUtil()
{
	gId('segutil').innerHTML = '<button class="btn btn-s" onclick="makeSeg()"><i class="icons btn-icon">&#xe18a;</i>Add segment</button>';
	for (var i=0; i<expanded.length; i++) if (expanded[i]) expand(i); // collapse all expanded elements
}
```

My populateSegments() is a bit different since I exposed power button and brightness slider. I would recommend doing the same in master.